presto-hive/src/main/java/com/facebook/presto/hive/HiveErrorCode.java

```diff
@@ -55,7 +55,8 @@
     HIVE_INVALID_BUCKET_FILES(28, EXTERNAL),
     HIVE_EXCEEDED_PARTITION_LIMIT(29, USER_ERROR),
     HIVE_WRITE_VALIDATION_FAILED(30, INTERNAL_ERROR),
-    HIVE_PARTITION_DROPPED_DURING_QUERY(31, EXTERNAL);
+    HIVE_PARTITION_DROPPED_DURING_QUERY(31, EXTERNAL),
+    HIVE_PARTITION_NOT_READABLE(32, USER_ERROR);
```

Comment:
This should be `HIVE_OBJECT_NOT_READABLE` since it applies to both tables and partitions. Or perhaps we should split this into two error codes, like we have for `HIVE_TABLE_OFFLINE` and `HIVE_PARTITION_OFFLINE `.